pkg/apiserver/request/requestinfo.go

```diff
@@ -103,7 +103,7 @@ type RequestInfoResolver struct {
 // /api
 // /healthz
 // /
-func (r *RequestInfoResolver) GetRequestInfo(req *http.Request) (RequestInfo, error) {
+func (r *RequestInfoResolver) GetRequestInfo(req *http.Request) (*RequestInfo, error) {
```

Comment:
Let's rename the function to `NewRequestInfo` or `BuildRequestInfo`. The source for confusion is the name, not the type.